src/main/java/com/example/imagescaler/upscale/apiprovider/UpscaleApiProviderImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.example.imagescaler.upscale.apiprovider;
+
+import com.example.imagescaler.upscale.configuration.ApiProperties;
+import java.time.Duration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+@Service
+public class UpscaleApiProviderImpl implements UpscaleApiProvider {
+  private final ApiProperties apiProperties;
+  private final WebClient webClient;
+
+  public UpscaleApiProviderImpl(
+    ApiProperties apiProperties,
```

Comment:
We usually call the configuration classes `Configuration`. Maybe ApiProperties (or ApiConfiguration) is too generic.

Since we're talking about the UpscaleApi, we could call it `UpscaleApiConfig.Configuration` instead of `ApiProperties`.